### PR TITLE
DAG: Fix ABI lowering with FP promote in strictfp functions

### PR DESCRIPTION
**Pull request by arsenm**
This was emitting non-strict casts in ABI contexts for illegal
types.

### CODE REVIEW

**Comment by arsenm** [comment]
ping

**Comment by arsenm** [comment]
ping

File: llvm/test/CodeGen/AMDGPU/strict_fpext.ll [hunk 1]
```diff
@@ -1,22 +1,47 @@
 ; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py UTC_ARGS: --version 2
 ; FIXME: Missing operand promote for f16
```
Comment (RKSimon) [inline llvm/test/CodeGen/AMDGPU/strict_fpext.ll:2]: Still need the FIXME?

Comment (arsenm) [reply llvm/test/CodeGen/AMDGPU/strict_fpext.ll:2]: Should have been removed in the previous commit to add the promotion handling

**Review by RKSimon — APPROVED** [review]
LGTM

**Pull request merged**